### ISSUE TITLE
Slow directory traversal

### ISSUE THREAD

**Issue by sirinsidiator** [ichaoX/ext-file#2]
Recursively scanning folders takes several times longer than with Chrome. Don't know if there is anything you can do about that, but I just wanted to mention it.

**Comment by ichaoX** [comment]
This is probably because methods like `FileSystemDirectoryHandle.entries()` and `FileSystemDirectoryHandle.values()` use `scandir` once and call `getKind` on demand.

I can think of two solutions at the moment:
1. prefetching and caching the results of `getKind`, but this can lead to unreliable results when cached.
2. change the scandir interface to include kind information, which may affect the performance of `FileSystemDirectoryHandle.keys()`.

However, these workarounds still don't match the performance of Chrome's native API.

**Comment by sirinsidiator** [comment]
I figured as much. I guess I can try to improve it in my case by making the directory scan a separate step that is rarely needed and additionally cache the file handles for the log files I am interested in, since they won't move anyway.

**Comment by sirinsidiator** [comment]
I changed my approach a bit and started to avoid making too many directory visits. The structure for the target folders is well known, so I can just stop visiting subfolders once I found what I am looking for in a directory tree.
As long as the user selects the intended folder, that makes it go from several seconds of scan time to below a second at the cost of not finding every single log file hidden away somewhere they are not supposed to be.
